plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java

```diff
@@ -108,4 +113,18 @@ public static void bindTablePropertiesProvider(Binder binder, Class<? extends Ta
     {
         tablePropertiesProviderBinder(binder).addBinding().to(type).in(Scopes.SINGLETON);
     }
+
+    @Singleton
+    @ForRecordCursor
+    @Provides
+    public ExecutorService createRecordCursorExecutor(CatalogName catalogName)
+    {
+        return newCachedThreadPool(daemonThreadsNamed(catalogName + "-record-cursor-%s"));
```

Comment:
why not use direct thread executor by default and only pass to different thread for connectors which actually need that?

Reply:
I believe all JDBC drivers need that, but I have no evidence. So we can either blindly fix all drivers (current code) or surely fix what we know is broken and where we know how to test it. I am fine either way.

Reply:
Oh. For some reason I assumed that ignoring interruption is not a common thing. And just a few JDBC drivers are **broken** that way. If that it common I am fine with having thread-pool as a default. It should not be a huge cost.

Reply:
See javadoc of `java.sql.Connection#setNetworkTimeout`

>    This method is intended to address a rare but serious
     * condition where network partitions can cause threads issuing JDBC calls
     * to hang uninterruptedly in socket reads, until the OS TCP-TIMEOUT
     * (typically 10 minutes).
     
It looks like even JDBC standard is aware of this problem.

Reply:
I think we should have a kill switch for this 

and yes, for it it would be sufficient to bind a direct executor